### DISCUSSION TITLE
Python Version Conflict

### DISCUSSION THREAD

**Discussion by marcelkahlen** [python-poetry/poetry#9172] [Q&A] [answered]
Dear all,

for my pyproject.toml, I restrict the python version with `python = "^3.10"`. This works for me for many of my python packages with mutliple additional dependencies.

However, when I add pyinstaller as a dependency, I get the following conflict on calling `poetry install`:
```
The current project's supported Python range (>=3.10,<4.0) is not compatible with some of the required packages Python requirement:
  - pyinstaller requires Python <3.13,>=3.8, so it will not be satisfied for Python >=3.13,<4.0

Because metrology-analyzer depends on pyinstaller (6.5.0) which requires Python <3.13,>=3.8, version solving failed.

  * Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties

    For pyinstaller, a possible solution would be to set the `python` property to ">=3.10,<3.13"
```
When I restrict restrict the python version to `python = ">=3.10,<3.13"` then it works. 

I do not understand why poetry is unable to solve the version problem with `python = "^3.10"` since this should also allow python versions `>=3.10,<3.13` which are suitable for pyinstaller.

I use python 3.11.2 and poetry 1.8.2. 

Thanks a lot for your help!
Marcel

**Comment by dimbleby** [comment]
as the error message says

> pyinstaller requires Python <3.13,>=3.8, so it will not be satisfied for Python >=3.13,<4.0

also please read the faq

**Reply by marcelkahlen** [reply]
I still do not see why poetry is not allowing the installation since my python version 3.11 satisfies the python restriction and the pyinstaller restriction.

**Reply by finswimmer** [reply, marked as answer]
https://python-poetry.org/docs/faq/#why-is-poetry-telling-me-that-the-current-projects-supported-python-range-is-not-compatible-with-one-or-more-packages-python-requirements

> Unlike pip, Poetry doesn’t resolve for just the Python in the current environment. Instead it makes sure that a dependency is resolvable within the given Python version range in pyproject.toml.

**Reply by marcelkahlen** [reply]
I see. Thank you both!
